### PR TITLE
Adding Wrapper for Ridge Solver

### PR DESCRIPTION
**Pull request by coreyhu**
Created a wrapper to run a sklearn.linear_model.Ridge solver in a Tensorflow Session to allow for faster convergence when optimizing a linear least squares + l2 regularization problem

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ekelsen, @rmlarsen: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by coreywho** [comment]
Hi, could someone check this PR? Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ekelsen, @rmlarsen: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ekelsen, @rmlarsen: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

File: tensorflow/contrib/opt/python/training/external_optimizer.py [hunk 1]
```diff
@@ -446,3 +447,99 @@ def _compute_gradients(tensor, var_list):
...
+      feed_dict: The training feed dict to be passed into `session.run`. 
+      **run_kwargs: kwargs to pass into `session.run`.
+    
+    """
+    session = session or ops.get_default_session()
+    feed_dict = feed_dict or {}
+
+    initial_values = [session.run(self.X, feed_dict=feed_dict), session.run(self.y, feed_dict=feed_dict)]
+
+    var_vals = self._minimize(
```
Comment (rmlarsen) [inline tensorflow/contrib/opt/python/training/external_optimizer.py]: this fits on a single line

File: tensorflow/contrib/opt/python/training/external_optimizer.py [hunk 2]
```diff
@@ -446,3 +447,99 @@ def _compute_gradients(tensor, var_list):
...
+class SklearnRidgeInterface:
+  """Wrapper allowing a 'sklearn.linear_model.Ridge' optimizer to operate in a 'tf.Session'.
+
+  This optimizer minimizes the linear least squares problem
+  with l2 regularization using the sklearn.linear_model.Ridge
+  solver. It minimizes the following loss:
+
+    argmin_{W, b} L(X, y) = (y - WX - b)^2 + l|W|
+
+  Where l is a hyperparameter.
```
Comment (rmlarsen) [inline tensorflow/contrib/opt/python/training/external_optimizer.py]: Can you please add a unit test for this Interface?

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by case540** [comment]
Seems like our bot mistakenly removed "waiting for response" label. Please address comments about tests on this PR. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 28 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
